Cargo.toml

```diff
@@ -10,47 +10,43 @@ rust-version = "1.65"
 keywords = [ "crypto", "signatures", "zero-knowledge", "bbs", "blind-signature" ]
 categories = [ "cryptography", "authentication" ]
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-
 [dependencies]
 rand = {version = "0.8.5"}
 zeroize = { version = "1.5" }
-serde = { version = "1.0.25", default-features = false, features = ["derive", "serde_derive"] }
-serde_json = "1.0.59"
-hex = "0.4.3"
-thiserror = "1.0.30"
+serde = { version = "1.0.25", default-features = false, features = ["alloc", "derive"] }
+hex = { version = "0.4.3", default-features = false, features = ["alloc"] }
+thiserror = {version = "1.0", default-features = false, package = "thiserror-nostd-notrait"}
```

Comment:
Is it possible to handle this as a feature? Forcing this to be `no_std` breaks compatibility with [identity.rs](https://github.com/iotaledger/identity.rs), which expects zkryptium errors to implement the StdError trait. This will cause issues when identity.rs is updated to support the new version.

PS: I also noticed that there is ongoing work on a `no_std` feature in the thiserror crate: https://github.com/dtolnay/thiserror/pull/304. So it is likely that in the near future it will be handled in this way.

Reply:
Ah, of course.  Helpfully, `thiserror-nostd-notrait` does still implement `std:error::Error` if you add its `std` feature.